File: Common/Entity/src/main/java/it/ey/dto/Sezione331DTO.java
package it.ey.dto;

import lombok.*;
import lombok.experimental.SuperBuilder;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder(toBuilder = true)
public class Sezione331DTO extends SezioneBaseDTO {

    private Long id;


    private  Long idPiao;


    private String contesto;
    private String descrizioneQualitativa;
    private String strategiaProgrammazione;
    private String obiettivoTrasformazione;
    private Boolean rimodulazione;
    private String strategiaCopertura;
    private String descrizioneStrategia;
    private String stimaEvoluzione;

    // NON esponiamo idStato
    private String statoSezione;
    private List<TabellaFunzionaleDTO> tabelleFunzionali;




}

File: dfp-piao-fe/src/app/shared/models/classes/sezione-331-dto.ts
import { TabellaFunzionaleDTO } from './tabella-funzionale-dto';

export class Sezione331DTO {
  id?: number;

  idPiao?: number;

  contesto?: string;
  descrizioneQualitativa?: string;
  strategiaProgrammazione?: string;
  obiettivoTrasformazione?: string;
  rimodulazione?: boolean;
  strategiaCopertura?: string;
  descrizioneStrategia?: string;
  stimaEvoluzione?: string;

  tabelleFunzionali?: TabellaFunzionaleDTO[];

  statoSezione?: string;
  campiModificati?: string;
  testoSezione?: string;
}
